tests/gateways/test_connection.py

```diff
@@ -272,9 +271,13 @@ def test_get_session_with_url_pattern(mocker, channel_settings_url, expect_match
         "conda.gateways.connection.session.get_channel_name_from_url",
         return_value=channel_url,
     )
-    mock_context: Context = mocker.patch("conda.gateways.connection.session.context")
-    mock_context.channel_settings = (
-        {"channel": channel_settings_url, "auth": "dummy_one"},
```

Comment:
This form of context mocking is too broad and results in the context object doing some unfortunate caching